charms/istio-pilot/src/charm.py

```diff
@@ -426,5 +513,40 @@ def _get_address_from_loadbalancer(svc):
         raise ValueError("Unknown situation - LoadBalancer service has no hostname or IP")
 
 
+def _validate_upgrade_version(versions) -> bool:
+    """Validates that the version of istioctl can upgrade the currently deployed Istio.
+
+    This asserts that the istioctl version is equal to or at most greater than the current Istio
+    control plane by no more than one minor version.
+
+    Args:
+        versions (dict): A dictionary containing:
+                            client: the client version (eg: istioctl)
+                            control_plane: the control plane version (eg: istiod)
+
+    Returns True if this is the case, else raises an exception with details.
+    """
+    client_version = Version(versions["client"])
+    control_plane_version = Version(versions["control_plane"])
+
+    if client_version < control_plane_version:
```

Comment:
I am confused about these comparisons. Where do you tell istioctl the version you want to upgrade to so that the client version and the control plane version are different?

Reply:
istioctl (the binary) only knows how to update to its own version.  So istioctl_1.16.2 knows how to upgrade to 1.16.2, nothing else.  Executing `istioctl version` returns the client (istioctl) and control_plane (whatever is in k8s) versions.  That means `istioctl version` tells us both what we are trying to upgrade to (client) and what we're upgrading from (control_plane)

Reply:
I think this is sorted, but following up with @dnplas offline.  Resolving for now